### DISCUSSION TITLE
Compatibility betweeen Sidekiq/Pro/Ent

### DISCUSSION THREAD

**Discussion by Tseyang** [sidekiq/sidekiq#5424] [Q&A]
Thanks for Sidekiq, its a really awesome piece of software that I (and my team) greatly appreciate. I'm not sure if there's actually a place to look this up, but I was wondering if there's a location to tell if versions of Sidekiq, Sidekiq-pro and Sidekiq-ent are compatible/incompatible? 

I'm trying to perform some version upgrades (currently Sidekiq 6.1 to 6.4) and was hoping on doing them independently but I'm not sure if I will cause regressions if I don't do some sort of lockstep upgrade across the packages.

If there isn't a straightforward way to check that versions are compatible, what is the recommended way to perform such upgrades?

Thanks in advance for the response and help.

**Comment by mperham** [comment]
Generally the idea is that the Sidekiq Ent or Pro gemspecs define the minimum versions required of their dependencies. Use `bundle up sidekiq-pro` or `bundle up sidekiq-ent` to upgrade your gem versions according to these specs.

The one thing you don't want to do is run `bundle up sidekiq` if you use a commercial version as that will leave you in a "half upgraded" state.
